integration-cli/docker_api_swarm_test.go

```diff
@@ -445,6 +445,43 @@ func (s *DockerSwarmSuite) TestApiSwarmServicesStateReporting(c *check.C) {
 		}
 	}
 }
+func (s *DockerSwarmSuite) TestApiSwarmLeaderElection(c *check.C) {
+	// Create 3 nodes
+	d1 := s.AddDaemon(c, true, true)
+	d2 := s.AddDaemon(c, true, true)
+	d3 := s.AddDaemon(c, true, true)
+
+	// pick out the leader
+	var leader *SwarmDaemon
+	for _, d := range []*SwarmDaemon{d1, d2, d3} {
+		if d1.getNode(c, d.NodeID).ManagerStatus.Leader {
```

Comment:
```
for _, n := range d1.listNodes() {
  for _, d := range []*SwarmDaemon{d1, d2, d3} {
    if d.NodeID == n.ID {
    }
  }
}
```

Not most important. I just looked first time that you were making separate requests to test the forwarding as well.